### PR TITLE
feat: executor supports callbacks with move-only types

### PR DESCRIPTION
**Pull request by antmat**
@3Hren @karitra PTAL

### CODE REVIEW

File: include/cocaine/api/executor.hpp [hunk 1]
```diff
@@ -6,6 +6,20 @@ namespace cocaine {
 namespace api {
 
 class executor_t {
+    template<class F>
+    class move_only_wrapper {
+    public:
+        move_only_wrapper(F _f):
+            f(new F(std::move(_f)))
```
Comment (3Hren) [inline include/cocaine/api/executor.hpp]: make_shared

File: include/cocaine/api/executor.hpp [hunk 2]
```diff
@@ -6,6 +6,20 @@ namespace cocaine {
...
+        {}
+
+        auto operator()() -> void {
```
Comment (3Hren) [inline include/cocaine/api/executor.hpp]: As I remember we leave `auto` on separate line.

File: include/cocaine/api/executor.hpp [hunk 3]
```diff
@@ -14,6 +28,12 @@ class executor_t {
 
     virtual
     auto spawn(work_t work) -> void = 0;
+
+    template<class F>
+    auto spawn(F f) -> typename std::enable_if<!std::is_copy_assignable<F>::value>::type {
+        std::function<void()> func(move_only_wrapper<F>(std::move(f)));
```
Comment (3Hren) [inline include/cocaine/api/executor.hpp]: Does intermediate variable introduction here make sense?

File: include/cocaine/api/executor.hpp (hunk 2 above)
Comment (antmat) [inline include/cocaine/api/executor.hpp]: not really sure) - https://github.com/3Hren/cocaine-core/blob/v0.12.11/src/service/runtime.cpp#L27

Comment (3Hren) [inline include/cocaine/api/executor.hpp]: Bah, I drop this tomorrow.

File: include/cocaine/api/executor.hpp (hunk 1 above)
Comment (antmat) [inline include/cocaine/api/executor.hpp]: I don't really see any andvantage here, but ok.

File: include/cocaine/api/executor.hpp (hunk 3 above)
Comment (antmat) [inline include/cocaine/api/executor.hpp]: nope

File: include/cocaine/api/executor.hpp (hunk 1 above)
Comment (3Hren) [inline include/cocaine/api/executor.hpp]: Using `make_shared` always have advantages comparing with `new`: https://herbsutter.com/2013/05/29/gotw-89-solution-smart-pointers/

**Comment by antmat** [comment]
fixed and rebased

**Pull request merged**